// react-native-wind/dist/styles/typography/font-size.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildFontSizes = exports.mergeFontSizes = exports.defaultFontSizes = void 0;
const customize_1 = require("../../core/customize");
exports.defaultFontSizes = {
    xs: {
        fontSize: 12,
        lineHeight: 16,
    },
    sm: {
        fontSize: 14,
        lineHeight: 20,
    },
    base: {
        fontSize: 16,
        lineHeight: 24,
    },
    lg: {
        fontSize: 18,
        lineHeight: 28,
    },
    xl: {
        fontSize: 20,
        lineHeight: 28,
    },
    '2xl': {
        fontSize: 24,
        lineHeight: 32,
    },
    '3xl': {
        fontSize: 26,
        lineHeight: 36,
    },
    '4xl': {
        fontSize: 36,
        lineHeight: 40,
    },
    '5xl': {
        fontSize: 48,
    },
    '6xl': {
        fontSize: 60,
    },
    '7xl': {
        fontSize: 72,
    },
    '8xl': {
        fontSize: 96,
    },
    '9xl': {
        fontSize: 128,
    },
};
const getCustomFontSizes = () => customize_1.customStylesDefined === null || customize_1.customStylesDefined === void 0 ? void 0 : customize_1.customStylesDefined.fontSize;
const mergeFontSizes = () => {
    const customFontSizes = getCustomFontSizes();
    fontSizes = Object.assign(Object.assign({}, exports.defaultFontSizes), customFontSizes);
};
exports.mergeFontSizes = mergeFontSizes;
let fontSizes;
const buildFontSizes = () => {
    const sizes = {};
    Object.keys(fontSizes).forEach((s) => {
        sizes[`text-${s}`] = fontSizes[s];
    });
    return sizes;
};
exports.buildFontSizes = buildFontSizes;
